Lib/test/test_asyncio/test_waitfor.py

```diff
@@ -256,7 +256,6 @@ async def inner():
         # Test that wait_for itself is properly cancellable
         # even when the initial task holds up the initial cancellation.
         task = asyncio.create_task(wait)
-        await asyncio.sleep(0.2)
```

Comment:
Curious why the test fails with this sleep.

Reply:
It is because `wait_for` no longer creates a new task for `coroutine`.

Reply:
Oh so it will still work with the sleep but the sleep is no longer necessary to make the test pass? Good call then!

Reply:
This test fails if the sleep is not removed.

When the sleep is removed, `task` was created and then cancelled immediately, meaning it did not get a chance to run, thus the `wait_for()` call inside it was never executed. I believe this change invalidates the test.

Reply:
It's too late here to investigate more but I take it that you believe the test is now passing for the wrong reason? @kumaraditya303 There's something to that...

Reply:
Perhaps I didn't make myself clear so here a longer explanation.

In previous implementation of `wait_for`, it worked by creating a new future and then awaiting on it so it was possible to cancel `wait_for` and would cancel that future but in the new implementation with `timeout` it no longer creates a new task so cancelling is has the same effect as cancelling the coroutine `wait_for` is waiting for but since here the coroutine is suppressing the cancellation so it will not raise `CancelledError`  with `sleep` but return 42. I can remove the test if you want though but went I went with making smallest possible change to tests to make it pass.

Reply:
Yes, I believe the change made to this test means it's not actually testing `wait_for()` anymore, and it passes for the wrong reason.

If I understand correctly, this test requires that when `wait_for(inner_coro(), timeout)` times out, even if `inner_coro` ignores cancellation (by catching and not reraising CancelledError), `wait_for` still raises CancelledError. This PR modifies this behavior, and returns `inner_coro`'s result instead.

Reply:
Okay I'll remove the test. 

> This PR modifies this behavior, and returns inner_coro's result instead.

Correct!, since no future is created, `wait_for` does what the coroutine does.

Reply:
I think we want to preserve existing behavior, especially if there's a test specifically testing for it? @gvanrossum should decide, I believe.

Reply:
The behavior does not make sense, if you `wait_for` a coroutine and then cancel it and if the coroutine suppresses the cancellation and returnssomething then `wait_for` should return it rather than throwing `CancelledError`. This is one of the bugs we are trying to fix here. It also aligns it with how `timeout` works.